web3/contract/utils.py

```diff
@@ -60,10 +66,50 @@
         AsyncWeb3,
         Web3,
     )
+    from web3.providers.persistent import (  # noqa: F401
+        PersistentConnectionProvider,
+    )
 
 ACCEPTABLE_EMPTY_STRINGS = ["0x", b"0x", "", b""]
 
 
+@curry
+def format_contract_call_return_data_curried(
```

Comment:
Hmm I'm not sure. It's really only the `return_data` that we want to be able to pass in at a later time, everything else is preconfigured in the curried method. Since `call_contract_function` doesn't take the `return_data` argument I don't see how it can replace this. Is that kind of what you meant?=

Reply:
oh, gotcha. don't mind me :)